src/java/io/bazel/rulesscala/test_discovery/DiscoveredTestSuite.scala

```diff
@@ -134,4 +138,27 @@ object PrefixSuffixTestDiscoveringSuite {
   private def innerClasses(testClassName: String): Boolean =
     testClassName.contains('$')
 
+  private def containsTests(testClass: Class[_]): Boolean =
+    annotatedWithRunWith(testClass) || hasTestAnnotatedMethodsInClassHierarchy(testClass)
+
+  private def annotatedWithRunWith(testClass: Class[_]) =
+    testClass.getAnnotation(runWithAnnotation) != null
+
+  @tailrec
+  private def hasTestAnnotatedMethodsInClassHierarchy(testClass: Class[_]): Boolean =
+    Option(testClass) match {
+      case None => false
+      case Some(currentTestClass) if hasTestAnnotatedMethodsInCurrentClass(currentTestClass) => true
+      case Some(currentTestClass) => hasTestAnnotatedMethodsInClassHierarchy(currentTestClass.getSuperclass)
+    }
+
+  private def hasTestAnnotatedMethodsInCurrentClass(testClass: Class[_]): Boolean =
+    testClass.getDeclaredMethods.exists { method =>
+      method.getAnnotations.exists { annotation: Annotation =>
+        testAnnotation.isAssignableFrom(annotation.annotationType)
+      }
+    }
+
+  private lazy val runWithAnnotation = classOf[RunWith]
```

Comment:
We only need it if we get to that phase. I had a different thought when I started but now it's probably just an over optimization. I'll drop it.